lib/msal-node/src/client/ManagedIdentityApplication.ts

```diff
@@ -0,0 +1,275 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    AccessTokenEntity,
+    AuthenticationResult,
+    Authority,
+    AuthorityOptions,
+    CacheManager,
+    ClientAuthErrorCodes,
+    Constants,
+    CredentialFilter,
+    CredentialType,
+    DEFAULT_CRYPTO_IMPLEMENTATION,
+    DefaultStorageClass,
+    INetworkModule,
+    Logger,
+    NetworkManager,
+    ProtocolMode,
+    ResponseHandler,
+    ScopeSet,
+    TimeUtils,
+    createClientAuthError,
+} from "@azure/msal-common";
+import { ServiceFabric } from "./ManagedIdentitySources/ServiceFabric";
+import { AppService } from "./ManagedIdentitySources/AppService";
+import { CloudShell } from "./ManagedIdentitySources/CloudShell";
+import { AzureArc } from "./ManagedIdentitySources/AzureArc";
+import { Imds } from "./ManagedIdentitySources/Imds";
+import {
+    ManagedIdentityConfiguration,
+    ManagedIdentityNodeConfiguration,
+    buildManagedIdentityConfiguration,
+} from "../config/Configuration";
+import { version, name } from "../packageMetadata.js";
+import { ManagedIdentityRequest } from "../request/ManagedIdentityRequest";
+import { ManagedIdentityId } from "../config/ManagedIdentityId";
+import { CryptoProvider } from "../crypto/CryptoProvider";
+
+/**
+ * Class to initialize a managed identity and identify the service
+ */
+export class ManagedIdentityApplication {
+    private managedIdentityId: ManagedIdentityId;
+
+    private config: ManagedIdentityNodeConfiguration;
+
+    private logger: Logger;
+    private cacheManager: CacheManager;
+    private networkClient: INetworkModule;
+    private networkManager: NetworkManager;
+    private cryptoProvider: CryptoProvider;
+
+    private identitySource:
+        | ServiceFabric
+        | AppService
+        | CloudShell
+        | AzureArc
+        | Imds;
+
+    private fakeAuthority: Authority;
+
+    constructor(
+        managedIdentityId: ManagedIdentityId,
+        configuration?: ManagedIdentityConfiguration
+    ) {
+        this.config = buildManagedIdentityConfiguration(configuration || {});
+        this.managedIdentityId = managedIdentityId;
+
+        this.logger = new Logger(
+            this.config.system.loggerOptions,
+            name,
+            version
+        );
+
+        this.cacheManager = new DefaultStorageClass(
```

Comment:
What is the difference between `NodeStorage`, which ClientApplicaiton uses and this one?

Reply:
Good question, I'll have to look into this more. It appears NodeStorage is used in TokenCache, and TokenCache's methods are exposed to the user. For example: serialize, deserialize, getAllAccounts, etc. I don't think we want to expose these to the developer of an MSI application? If not, CacheManager should suffice for storing tokens in a cache. @sameerag am I understanding this correctly?

Reply:
On second pass, it looks like NodeStorage extends CacheManager, which this uses. 

NodeStorage "implements Storage for node, reading cache from user specified storage location or an  extension library". Managed Identity doesn't do either of those thing, so CacheManager should suffice.

Reply:
Another Update. I made a mistake when first using DefaultStorageClass. None of the cache methods are implemented in DefaultStorageClass. I should have been using NodeStorage initially.